src/hotspot/share/runtime/flags/jvmFlagConstraintsCompiler.cpp

```diff
@@ -439,3 +439,29 @@ JVMFlag::Error ControlIntrinsicConstraintFunc(ccstrlist value, bool verbose) {
   return JVMFlag::SUCCESS;
 }
 
+JVMFlag::Error InlineCacheBufferSizeConstraintFunc(int value, bool verbose) {
+  if (value <= 0) {
+      JVMFlag::printError(verbose,
+                          "InlineCacheBufferSize (%d) must be "
+                          "greater than %d\n",
+                          value, 0);
+      return JVMFlag::VIOLATES_CONSTRAINT;
+  }
+
+  if ((uintx)value >= NonNMethodCodeHeapSize) {
+    JVMFlag::printError(verbose,
+                        "InlineCacheBufferSize (%d) must be "
+                        "less than NonNMethodCodeHeapSize (" UINTX_FORMAT ")\n",
+                        value, NonNMethodCodeHeapSize);
+    return JVMFlag::VIOLATES_CONSTRAINT;
+  }
+
+  if ((value % CodeEntryAlignment) != 0) {
```

Comment:
I don't understand why this is necessary.

Reply:
It is needed to avoid 'failed: _buffer_size  not aligned' crashes on debug builds: 
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/code/stubs.cpp#L221

Reply:
That sounds like a bug.  We already align the code_begin().  I see no reason to align code_end() as well.  It just wastes space.

Reply:
OK, can I remove these guarantees for _buffer_size and _buffer_limit in this PR? Or do I need to create a separate one?

Reply:
I would like to see it cleaned up in this PR.

Reply:
I changed the InlineCacheBufferSize type to size_t and removed unnecessary guarantees for _buffer_size and _buffer_limit.
@dean-long could you please take a look?